src/main/java/org/sagebionetworks/web/client/presenter/ProfilePresenter.java

```diff
@@ -390,20 +387,34 @@ public void refreshTeams() {
 		view.showTeamsLoading();
 		teamNotificationCount = 0;
 		view.clearTeamNotificationCount();
-		if (isOwner)
+		if (isOwner) {	
 			view.refreshTeamInvites();
-		AsyncCallback<List<Team>> teamCallback = new AsyncCallback<List<Team>>() {
-			@Override
-			public void onFailure(Throwable caught) {
-				view.setTeamsError(caught.getMessage());
-			}
-			@Override
-			public void onSuccess(List<Team> teams) {
-				view.setTeams(teams,isOwner);
-			}
-		};
-		
-		TeamListWidget.getTeams(currentUserId, synapseClient, adapterFactory, teamCallback);
+			AsyncCallback<List<TeamRequestBundle>> teamCallback = new AsyncCallback<List<TeamRequestBundle>>() {
+				@Override
+				public void onFailure(Throwable caught) {
+					view.setTeamsError(caught.getMessage());
+				}
+				@Override
+				public void onSuccess(List<TeamRequestBundle> teams) {
+					view.setTeamsFromBundle(teams);
+				}
+			};
+			TeamListWidget.getTeamBundles(currentUserId, synapseClient, adapterFactory, teamCallback);
+		} else {
+			AsyncCallback<List<Team>> teamCallback = new AsyncCallback<List<Team>>() {
+				@Override
+				public void onFailure(Throwable caught) {
+					view.setTeamsError(caught.getMessage());
+				}
+				@Override
+				public void onSuccess(List<Team> teams) {
+					view.setTeams(teams);
+				}
+			};
+			
+			TeamListWidget.getTeams(currentUserId, synapseClient, adapterFactory, teamCallback);

```

Comment:
TeamListWidget.getTeams is only called from ProfilePresenter.  Please remove completely.  Also remove SynapseClient.getTeamsForUser()